### DISCUSSION TITLE
Pins on Demand

### DISCUSSION THREAD

**Discussion by Crabmeat1432** [szhu/pushpin#4] [Q&A]
Hi,  I don't see a Pins on Demand Feature in PushPin, one of my favorite apps.  I'd like to PIN pushpin to my browser so it is always visible but don't want it to load the pins until I click on it, for a little more level of privacy.   It looks like this

Open blank Brower with no pins
Click pushpin
Pins are loaded into the browser

Sometimes I have to do this because I happened to open another instance of  Chrome, without the pins, and closed the pinned browser, then closed the unpinned browser.  When this happens a new browser instance opens with no pins - yay!   Most times though, I don't open a 2nd instance and the pins are there when I close and reopen.  This isn't ideal, for privacy reasons.  I'd prefer to load them just on demand every time.  Am I missing something?

**Comment by szhu** [comment]
Hi, thanks for your question! I think I understand what your question is but I'm not 100% sure, so I'd just like to confirm first.

Are you saying thats sometimes when:
1. You have only one Chrome window, and it has pinned tabs
2. You close the window
3. You open a new Chrome window

That new Chrome window has pinned tabs, but you wish it didn't?

---

If I've accurately understood your question, the answer is that this is built-in Chrome behavior! You can verify this by disabling the extension and trying it again.

I can add functionality in Pushpin to disable this behavior, but I think that would be out of scope. I believe the main users of Pushpin are users who like this functionality of Chrome and are annoyed when it doesn't happen when you close the pinned window before the un-pinned window! For example, see the users here: https://superuser.com/q/405345 These complaints are why I created Pushpin.

Let me know if I'm understanding the situation correctly, happy to hear your thoughts.

**Comment by Crabmeat1432** [comment]
Yes, that’s right - I’m always thinking about security - waiting for a day my computer locks up and I must bring it in.  I’m signed into many of those pins always, like messenger, Facebook, and Reddit.   I don’t want some computer guy going into my stuff when he opens the browser.  I know it’s not a guarantee, but he probably would not think to if they weren’t presented to him on the browser opening.  It’s not likely he’s going to poke around the browser extensions, but I can easily add them every time because I know the extension is there. 

 

Mike

**Comment by szhu** [comment]
I see what you mean, thanks for clarifying.

I think for your use case, there would be no way for me to add a feature to solve this with this extension.

The "normal way" to solve this problem is not available: I am not aware of an API that lets us change (or even read) the list of "saved pinned tabs" that Chrome will reopen when you start Chrome.

A workaround solution would be to listen to the `tabs.onCreated` event and quickly close all the pinned tabs.

However, in the example you described, I believe this will be too little too late: the hypothetical computer guy will have seen the pinned tabs quickly flash for a fraction of a second, tipping them off that there is snoopable content. Then all they have to do is look at your browser history and the pinned tabs will be right on top.

Here are two solutions to the problem you describe:
- Turn on full disk encryption (FileVault/BitLocker).
- Visit a computer service center that allows you to be present when they perform the repair.

**Comment by Crabmeat1432** [comment]
Thanks for looking into it Sean. It is very astute of you!

**Comment by Crabmeat1432** [comment]
I found a solution with this extension.

I can close all tabs before I shut Chrome down and  then, when chrome opens again with no tabs showing, and I can activate them again with your extension which Is not pinned to the search bar but sits inconspicuously invisible, unpinned, until needed..
